### PR TITLE
Bunch of nitpicks

### PR DESCRIPTION
**Pull request by emilio**
I just noticed one while writing #14719, and then grepped and couldn't stop.

r? @nox

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14728)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -725,10 +722,8 @@ impl ScriptThread {
 
         for (id, document) in self.documents.borrow().iter() {
             // Only process a resize if layout is idle.
-            let resize_event = document.window().steal_resize_event();
-            match resize_event {
-                Some((size, size_type)) => resizes.push((id, size, size_type)),
-                None => ()
+            if let Some((size, size_type)) = document.window().steal_resize_event() {
+                resizes.push((id, size, size_type))
```
Comment (Wafflespeanut) [inline components/script/script_thread.rs]: Nit: Trailing semicolon?

File: components/net_traits/lib.rs [hunk 2]
```diff
@@ -438,24 +438,17 @@ impl Metadata {
 
     /// Extract the parts of a Mime that we care about.
     pub fn set_content_type(&mut self, content_type: Option<&Mime>) {
-        match self.headers {
-            None => self.headers = Some(Serde(Headers::new())),
-            Some(_) => (),
+        if self.headers.is_none() {
+            self.headers = Some(Serde(Headers::new()))
```
Comment (Wafflespeanut) [inline components/net_traits/lib.rs]: Nit: Trailing semicolon here too?

**Review by ghost — APPROVED** [review]
Looks easy enough :smile:

r=me after addressing the comments.

File: components/net_traits/lib.rs (hunk 2 above)
Comment (emilio) [inline components/net_traits/lib.rs]: Looks more straight-forward and easy to read this way IMO.

File: components/profile/time.rs [hunk 3]
```diff
@@ -296,12 +296,7 @@ impl Profiler {
     }
 
     fn find_or_insert(&mut self, k: (ProfilerCategory, Option<TimerMetadata>), t: f64) {
-        match self.buckets.get_mut(&k) {
-            None => {},
-            Some(v) => { v.push(t); return; },
-        }
-
-        self.buckets.insert(k, vec!(t));
+        self.buckets.entry(k).or_insert_with(Vec::new).push(t)
```
Comment (emilio) [inline components/profile/time.rs]: Sure, if you feel strong about it... I don't personally, but fine

File: components/script/script_thread.rs (hunk 1 above)
Comment (emilio) [inline components/script/script_thread.rs]: Same reply as above.

File: components/profile/time.rs (hunk 3 above)
Comment (Wafflespeanut) [inline components/profile/time.rs]: I don't really have an argument. Even though `rustc` doesn't care about these cases, I thought we should terminate expressions with semicolons, and leave it as optional for keywords. I'll better leave this to you :)

**Comment by emilio** [comment]
Addressed review comments, thanks for the review @Wafflespeanut!

@bors-servo r=Wafflespeanut

**Pull request merged**